### ISSUE TITLE
PSL- Cable Status shows plugged in, but it is not (when the Pod Point is in Smart mode)

### ISSUE THREAD

**Issue by busywait** [mattrayner/pod-point-home-assistant-component#55]
## Version of the custom_component
2.0.1 
(This issue was not seen previously in the 1.X versions)

## Configuration

I have the original big round Pod Point, firmware A22P-2.35.0-00002 reported by this integration.

```yaml
My whole configuration.yaml? Not relevant here.
```

## Describe the bug
Since I updated to the new 2.0.1 PodPoint integration my "Cable Status" shows Plugged In, but it is not. I have a Solo 1

I have "Smart Charge" mode turned on (schedules set via the Pod Point mobile app).

Plugging the connector in to my car, wating 5 minutes the state remains Plugged In.

Removing the Pod Point connector from my car and waiting another 5 minutes, the Cable Status remains Plugged In.

I expect Connection Pod Point and Connection Car history to be approximately the same (bar 2 and 3 below)
![Screenshot_20240411_162845_Home Assistant](https://github.com/mattrayner/pod-point-home-assistant-component/assets/3671165/7e00b011-888b-4a78-9c8d-ec05d96671bd)

I updated the Pod Point integration around 10 am on the screenshot above.

Starting a Schedule Override puts the Cable Status to "Unplugged" for this time.

Stopping the Schedule Override using the Stop Pod Point Charge Now service puts the Cable Status back to  "Plugged In"

## Debug log

Selected Cable Status attributes from the Home Assistant Developer Tools tab:
```text

statuses:
  - id: 1
    name: Available
    key_name: available
    label: Available
    door: A
    door_id: 1
  - id: 99
    name: SUSPENDED_EVSE
    key_name: SUSPENDED_EVSE
    label: SUSPENDED_EVSE
    door: A
    door_id: 1
unit_connectors:
  - connector:
      id: 102
      door: A
      door_id: 1
      power: 7
      current: 32
      voltage: 230
      charge_method: Single Phase AC
      has_cable: true
      socket:
        type: IEC 62196-2 Type 2
        description: Type 2 socket
        ocpp_name: sType2
        ocpp_code: 3


  version_info:
    manifest_id: A22P-2.35.0-00002
  update_status:
    is_update_available: false
charge_override: null
offering_energy: false
last_message_at: "2024-04-11T15:40:44+00:00"
charging_state: SUSPENDED_EVSE
connectivity_status:
  ppid: PSL-16XXXX
  connected_components:
    - evses
  evses:
    - id: 1
      architecture: arch2
      connectivityState:
        protocol: PPCP
        connectivityStatus: ONLINE
        signalStrength: -75
        lastMessageAt: "2024-04-11T15:40:44+00:00"
        connectionStartedAt: null
        connectionQuality: 2
      connectors:
        - id: 1
          door: A
          chargingState: SUSPENDED_EVSE
      energyOfferStatus:
        isOfferingEnergy: false
        reason: CHARGE_SCHEDULE
        until: "2024-04-12T00:00:00+00:00"
        randomDelay: null
        doNotCache: false
state: suspended-evse
device_class: plug
friendly_name: PSL-166021 Cable Status


```

**Comment by mattrayner** [comment]
Hi @busywait thanks for your report, it seems that the API is doing something a little strange here.

As there is no documentation for any of the Pod Point APIs, the behaviour we implement is through observation.

The section of the API response which is causing your particular problem is this:

```yaml
statuses:
  - id: 1
    name: Available
    key_name: available
    label: Available
    door: A
    door_id: 1
  - id: 99
    name: SUSPENDED_EVSE
    key_name: SUSPENDED_EVSE
    label: SUSPENDED_EVSE
    door: A
    door_id: 1
```

The two statuses here come from two different endpoints.

The 'pod' endpoint is returning the status 'available' as shown here:

```yaml
  - id: 1
    name: Available
    key_name: available
    label: Available
    door: A
    door_id: 1
```

There is a 'new' endpoint we are consuming called 'connectivity status' which (from observation) appears to show additional information about your pod's state. This endpoint is returning 'suspended_evse' as the status.

```yaml
  - id: 99
    name: SUSPENDED_EVSE
    key_name: SUSPENDED_EVSE
    label: SUSPENDED_EVSE
    door: A
    door_id: 1
```

Until now, I though this would be impossible, as 'Available' is interpreted as:

> This pod is 'available' to be used for charging (a hold over from the public network where a pod is either available, charging, or out of service)

Conversely, 'Suspended EVSE' is interpreted as the following:

> You pod is not delivering power to the connected EV as the session is 'paused' (there is a schedule in place, or a problem with the connection to the EV).

It appears that my interpretation of 'suspended_evse' is fundamentally flawed.

> During my observation the only time I was able to cause 'suspended_evse' was by partially disconnecting my pod-side cable connection and connecting my EV. This causes an 'out-of-service' status for the pod whilst the door status showed as 'charging'.

Based on my one scenario I wrote some logic which said something like 'If sustepended-evse, cable is connected'.

I will now write some new logic which is: 

> If suspended-evse, check door status, if door status is available, cable is not connected, if door status is charging, cable is connected

I will look to get a new version of the integration (2.0.2) out shortly which should cater for this scenario!

**Comment by mattrayner** [comment]
@busywait I have released a new version of the integration (2.0.2) which should fix this issue

**Comment by busywait** [comment]
> I have released a new version of the integration (2.0.2) which should fix this issue

Thanks - I've installed and restarted. The Cable Status (unplugged, plugged in) now matches the real world :)

The main status changes from "Waiting for schedule" to "Charging Paused" when I plug the car in and there is a schedule active that prevents charging. Previously that situation showed the status to "Connected but waiting for schedule". Is Connected but waiting for schedule still a possible state now? ("Connected but waiting for schedule" *is* still mentioned in the release notes). I will mentally translate Charging Paused to "Not charging (Pod Point settings)".

When I connected the car while the Pod Point is available but charging is prevented by limits in the car then the Pod Point status showed "Charged" which is helpful and makes sense. I understand that as "Not charging (car settings)".

BTW, is the capital P in "Charging *P*aused" deliberate?

**Comment by mattrayner** [comment]
So, the 'connected but waiting for schedule' status is inferred by the integration and doesn't come directly from the pod point APIs, they only show 'connected'.

The logic on the connected but waiting for schedule is something like:

> If the current date/time is within a schedule window AND the pod is connected, show 'connected but waiting for schedule'

This, however is a bit one dimensional, it only cares about the schedule, but not other resons that the charger may not be delivering power, including the pod pausing printing due to power balancing in your home, being turned off via the key underneath, or other reasons the Pod Point determine.

So, the new 'Charging paused' status (It shouldnt have the uppercase 'P') comes in. Its determined from a new `connectivity_status` endpoint found in the app, and is slightly wider in its scope.

Now, can your pod still be 'connected but waiting for schedule' now that 'Charging paused' has been implemented? **Absolutely!** the 'original' status was determined directly using date from the pod api, including the status of the 'door' and the schedules in effect, for this reason it **should** always be possible to determine the 'connected but waiting for schedule' status. The newer 'Charging paused' status requires an additional API call which _may_ fail, and the response implies that now all devices will return it, its an API with a 'capabilites' key, and then additonal data.

Again, all of this is inferred from how the mobile app works, and through observation as there is no documentation for any of the Pod Point APIs

I hope this helps!

**Comment by mattrayner** [comment]
Spelling issue fixed in 2.0.3 just released
